### ISSUE TITLE
AppCheck doesn't work on iOS app running on Mac (Apple Silicon)

### ISSUE THREAD

**Issue by diomidispapas** [firebase/firebase-ios-sdk#10138]
### Description

I have integrated AppCheck framework in our app and it works as expected on iOS, iPadOS but not when we run the iOS app on an Apple Silicon Mac (M1 / M1 Pro). A token is not generated 

### Reproducing the issue

Try creating an app check token on production environment from an iOS running on Mac

### Firebase SDK Version

8.10.0

### Xcode Version

13.2.1

### Installation Method

Carthage

### Firebase Product(s)

App Check

### Targeted Platforms

iOS, macOS

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml
<!--- Paste the contents of your Podfile.lock *inside* this code block.
  This will be automatically formatted into code.
--->



```

</details>

**Comment by morganchen12** [comment]
I think this is a duplicate of #10095.

**Comment by andrewheard** [comment]
Hi @diomidispapas, are you using App Attest or DeviceCheck as your token provider? I recently saw this note in Apple's [documentation](https://developer.apple.com/documentation/devicecheck/dcappattestservice/3573915-issupported):

> If you read isSupported from an app running on a Mac device, the value is false. This includes Mac Catalyst apps, and iOS or iPadOS apps running on Apple silicon.

I think @morganchen12 is right that it's a duplicate of #10095. I don't have an Apple Silicon Mac in front of me at the moment to test it but I tried running a sample Mac Catalyst app on my Intel machine (macOS 12.5) and DeviceCheck is working but App Attest is not. As a potential workaround, you could consider falling back to DeviceCheck (more widely supported) on devices that don't support App Attest.

**Comment by diomidispapas** [comment]
DeviceCheck seem to work on iOS apps running on Apple silicon Macs. 
We ended up doing: 

`      if #available(iOS 14.0, *), !isiOSAppOnMac {
            return AppAttestProvider(app: app)
        } else {
            return DeviceCheckProvider(app: app)
        }`
